Tools/i18n/pygettext.py

```diff
@@ -340,13 +340,21 @@ def __waiting(self, ttype, tstring, lineno):
                 elif ttype not in (tokenize.COMMENT, tokenize.NL):
                     self.__freshmodule = 0
                 return
-            # class docstring?
-            if ttype == tokenize.NAME and tstring in ('class', 'def'):
-                self.__state = self.__suiteseen
+            # class or func/method docstring?
+            if ttype == tokenize.NAME:
+                if tstring == 'def':
+                    self.__state = self.__funcseen
+                elif tstring == 'class':
+                    self.__state = self.__suiteseen
                 return
         if ttype == tokenize.NAME and tstring in opts.keywords:
             self.__state = self.__keywordseen
 
+    def __funcseen(self, ttype, tstring, lineno):
+        # ignore anything until we see the closing parenthesis
+        if ttype == tokenize.OP and tstring == ')':
```

Comment:
What about the following example?
```python
def foo(bar=()):
```

Reply:
That's a very good point. I will rethink my approach